tests/rptest/tests/pandaproxy_test.py

```diff
@@ -333,6 +334,126 @@ def _create_named_consumer(self,
+            assert sc_res.status_code == requests.codes.no_content
+
+            # Maybe set consumer offsets to 0
+            parts = [0, 1, 2]
+            if do_set_offsets:
+                sco_req = dict(partitions=[
+                    dict(topic=topic_name, partition=p, offset=expected_offset)
+                    for p in parts
+                ])
+                co_res_raw = c0.set_offsets(data=json.dumps(sco_req),
+                                            auth=auth_tuple)
+                assert co_res_raw.status_code == requests.codes.no_content
+
+            self.logger.debug(f"Check consumer offsets")
+            co_req = dict(partitions=[
+                dict(topic=topic_name, partition=p) for p in parts
+            ])
+            offset_result_raw = c0.get_offsets(data=json.dumps(co_req),
+                                               auth=auth_tuple)
+            assert offset_result_raw.status_code == requests.codes.ok
+            res = offset_result_raw.json()
+            # Should be one offset for each partition that we previously set offsets for
+            assert len(res["offsets"]) == len(parts)
+            for r in res["offsets"]:
+                assert r["topic"] == topic_name
+                assert r["partition"] in parts
+                assert r["offset"] == expected_offset
+                assert r["metadata"] == ""
+
+        data = '''
+        {
+            "records": [
+                {"value": "dmVjdG9yaXplZA==", "partition": 0},
+                {"value": "cGFuZGFwcm94eQ==", "partition": 1},
+                {"value": "bXVsdGlicm9rZXI=", "partition": 2}
+            ]
+        }'''
+
+        self.logger.info(f"Producing to topic: {topic_name}")
+        check_produce_output(self._produce_topic(topic_name,
+                                                 data,
+                                                 auth=auth_tuple),
+                             expected_offset=0)
+
+        self.logger.info(f"Fetch from topic: {topic_name}")
+        check_fetch_output(self._fetch_topic(topic_name, 0, auth=auth_tuple),
+                           topic_name=topic_name,
+                           expected_data=json.loads(data),
+                           expected_offset=0)
+
+        self.logger.info("Check consumer offsets")
+        group_id = f"pandaproxy-group-{uuid.uuid4()}"
+        check_offsets(group_id=group_id,
+                      topic_name=topic_name,
+                      expected_offset=0,
+                      auth_tuple=auth_tuple)
+
+        self.logger.debug("Restart the http proxy")
+        admin = Admin(self.redpanda)
+        for node in self.redpanda.nodes:
+            result_raw = admin.redpanda_services_restart(
+                rp_service='http-proxy', node=node)
+            check_service_restart(self.redpanda, "Restarting the http proxy")
+            self.logger.debug(result_raw)
+            assert result_raw.status_code == requests.codes.ok
```

Comment:
@BenPope  We need to tell each broker to restart the PP since they each hold their own instance correct?
Perhaps a follow-up ticket could be adding a mechanism for each PP/SR instance to ping each other?

Reply:
This feature is a last resort to reset a broken PP without restarting the broker, it's possible that they all break in the same way at the same time, but it's not necessarily the case. Resetting PP is quite drastic, I think it's ok to target just one at a time.